driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -120,31 +124,33 @@ public void run() {
 
         Executor initExecutor = manager.cluster.manager.configuration.getPoolingOptions().getInitializationExecutor();
 
-        ListenableFuture<List<Void>> allConnectionsFuture = Futures.allAsList(connectionFutures);
+        ListenableFuture<List<Void>> allConnectionsFuture = Futures.successfulAsList(connectionFutures);
 
         final SettableFuture<Void> initFuture = SettableFuture.create();
-        Futures.addCallback(allConnectionsFuture, new FutureCallback<List<Void>>() {
+        Futures.addCallback(allConnectionsFuture, new MoreFutures.SuccessCallback<List<Void>>() {
             @Override
             public void onSuccess(List<Void> l) {
+                // Some of the connections might have failed, keep only the successful ones
+                ListIterator<Connection> it = connections.listIterator();
```

Comment:
Playing with this some more and I think maybe we do need to change the behavior, consider this scenario:   You have a 2 node cluster with node1 being up and running and node 2 being down.   You create a Cluster object with node1 being your contact point.   node2 is discovered in system.peers so we create a pool to it.  All connections fail, yet the host remains in the 'UP' state.  Wouldn't that be a problem?  We'd keep trying to borrow connections on node2 and would get exceptions.

A test that effectively does this:  RecommissionedNodeTest#should_ignore_recommissioned_node_on_control_connection_reconnect.